csp-mixin/panels.libsonnet

```diff
@@ -842,5 +842,312 @@ local commonlib = import 'common-lib/common/main.libsonnet';
+      + g.panel.timeSeries.standardOptions.withOverrides([]),
+
+    avm_cpu_utilization:
+      this.signals.virtualMachines.cpuUtilization.asTimeSeries()
+      + commonlib.panels.generic.timeSeries.base.stylize()
+      + g.panel.timeSeries.standardOptions.withUnit('percent')
+      + self._avm_timeSeriesCommon(),
+
+    avm_available_memory:
+      this.signals.virtualMachines.availableMemory.asTimeSeries()
+      + commonlib.panels.generic.timeSeries.base.stylize()
+      + g.panel.timeSeries.standardOptions.withUnit('decbytes')
+      + self._avm_timeSeriesCommon(),
+
+    avm_cpu_credits_consumed:
+      this.signals.virtualMachines.cpuCreditsConsumed.asTimeSeries()
+      + commonlib.panels.generic.timeSeries.base.stylize()
+      + g.panel.timeSeries.standardOptions.withUnit('short')
+      + self._avm_timeSeriesCommon(),
+
+    avm_cpu_credits_remaining:
+      this.signals.virtualMachines.cpuCreditsRemaining.asTimeSeries()
+      + commonlib.panels.generic.timeSeries.base.stylize()
+      + g.panel.timeSeries.standardOptions.withUnit('short')
+      + self._avm_timeSeriesCommon(),
+
+    avm_disk_total_bytes:
+      this.signals.virtualMachinesOverview.diskReadBytes.asTimeSeries()
+      + commonlib.panels.generic.timeSeries.base.stylize()
+      + this.signals.virtualMachinesOverview.diskWriteBytes.asPanelMixin()
+      + g.panel.timeSeries.standardOptions.withOverrides([])
+      + g.panel.timeSeries.standardOptions.withUnit('decbytes')
+      + self._avm_timeSeriesCommon(),
+
+    avm_disk_operations:
+      this.signals.virtualMachinesOverview.diskReadOperations.asTimeSeries()
+      + commonlib.panels.generic.timeSeries.base.stylize()
+      + this.signals.virtualMachinesOverview.diskWriteOperations.asPanelMixin()
+      + g.panel.timeSeries.standardOptions.withOverrides([])
+      + g.panel.timeSeries.standardOptions.withUnit('cps')
+      + self._avm_timeSeriesCommon(),
+
+    avm_network_total:
+      this.signals.virtualMachinesOverview.networkInTotal.asTimeSeries()
+      + commonlib.panels.generic.timeSeries.base.stylize()
+      + this.signals.virtualMachinesOverview.networkOutTotal.asPanelMixin()
+      + g.panel.timeSeries.standardOptions.withOverrides([])
+      + g.panel.timeSeries.standardOptions.withUnit('decbytes')
+      + self._avm_timeSeriesCommon(),
+
+    avm_connections:
+      this.signals.virtualMachinesOverview.inboundFlows.asTimeSeries()
+      + commonlib.panels.generic.timeSeries.base.stylize()
+      + this.signals.virtualMachinesOverview.outboundFlows.asPanelMixin()
+      + g.panel.timeSeries.standardOptions.withOverrides([])
+      + g.panel.timeSeries.standardOptions.withUnit('cps')
+      + self._avm_timeSeriesCommon(),
+
+    avm_network_in_by_instance:
+      this.signals.virtualMachines.networkInByVM.asTimeSeries()
+      + commonlib.panels.generic.timeSeries.base.stylize()
+      + g.panel.timeSeries.standardOptions.withUnit('decbytes')
```

Comment:
nit: you can just set units in signal itself. It will be 'propogated' to any panel where you would render that signal with functions like asTimeSeries(), asStat(), asPanelMixin(), asTableColumn() and so on.